src/hosts.cpp

```diff
@@ -85,6 +86,37 @@ code hosts::fetch(address& out) const
     ///////////////////////////////////////////////////////////////////////////
 }
 
+code hosts::fetch(address::list& out) const
+{
+    if (disabled_)
+        return error::not_found;
+
+    // Critical Section
+    ///////////////////////////////////////////////////////////////////////////
+    shared_lock lock(mutex_);
+
+    if (stopped_)
+        return error::service_stopped;
+
+    if (buffer_.empty())
+        return error::not_found;
+
+    const size_t count = std::min(buffer_.size(),
```

Comment:
buffer_.size() should never be greater than host_pool_capacity, so this always returns the former.